https://github.com/miurahr/pyppmd/pull/133.patch

From 95f77d9d0f4c3cabaf5a97da7f0cc74aaf582459 Mon Sep 17 00:00:00 2001
From: Mark Mentovai <mark@mentovai.com>
Date: Sat, 24 Oct 2026 16:54:14 -0400
Subject: [PATCH] Fix assertion in OutputBuffer_Grow to permit rest == 0

A comment in this function indicates that (rest == 0) is possible, but
an assertion insisted that (rest > 0). This was likely not noticed
because standard Python module builds define NDEBUG, circumventing
assertions. With NDEBUG undefined, this assertion was tripped by
test_ppmd7.py test_ppmd7_decode_chunks. Relax the assertion to tolerate
(rest >= 0).
---
 src/lib/buffer/blockoutput.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/lib/buffer/blockoutput.h src/lib/buffer/blockoutput.h
index 25d74cf..c5cc210 100644
--- src/lib/buffer/blockoutput.h
+++ src/lib/buffer/blockoutput.h
@@ -159,7 +159,7 @@ OutputBuffer_Grow(BlocksOutputBuffer *buffer, OutBuffer *ob) {
     if (buffer->max_length >= 0) {
         /* If (rest == 0), should not grow the buffer. */
         Py_ssize_t rest = buffer->max_length - buffer->allocated;
-        assert(rest > 0);
+        assert(rest >= 0);
 
         /* block_size of the last block */
         if (block_size > rest) {
